### PR TITLE
Fixed Wasdk breaking on net7+

### PR DESCRIPTION
**Pull request by Arlodotexe**
This PR ensures compatibility for `net7.0-windows10.0.19041.0` consumers when using the WinAppSDK head, addressing potential issues they might encounter due to TargetFramework mismatches.

### Background
NuGet's handling of TargetFrameworks for packages can lead to situations where, despite compatibility, the desired TFM might not be selected by the package manager.

For instance, while a project targeting `net7.0-windows10.0.19041.0` _can_ use packages targeting `net6.0-windows10.0.19041.0`, NuGet's TFM precedence might pick `net7.0` over `net6.0-windows10.0.19041.0` when the exact match isn't available.

Previously, it was believed that the solution might require downgrading some Uno libraries from net7.0 to netstandard2.0. However, further investigations highlighted that this wouldn't be necessary.

### Changes made in this PR:

- Addition of multi-targeting for WinAppSDK components, allowing them to target both `net6.0-windows10.0.19041.0` and `net7.0-windows10.0.19041.0`.
- Refactoring of certain conditions to correctly identify and work with the specified target frameworks.

### CODE REVIEW

**Comment by Arlodotexe** [comment]
Related - https://github.com/CommunityToolkit/Windows/issues/186

File: ProjectHeads/App.Head.props [hunk 1]
```diff
@@ -9,7 +9,7 @@
   </ItemGroup>
 
   <!-- Source generators -->
-  <ItemGroup Condition="'$(TargetFramework)' != '$(WinAppSdkTargetFramework)'">
+  <ItemGroup Condition="'$(IsWinAppSdk)' != 'true'">
```
Comment (michael-hawker) [inline ProjectHeads/App.Head.props:12]: I thought we couldn't use these props for these types of references and have to rely on the TFMs, so shouldn't this switch to the `Contains` like you did above?

Comment (Arlodotexe) [reply ProjectHeads/App.Head.props:12]: You're right, project heads don't use the MultiTarget system in quite the same way library projects do, and they don't have access to the `MultiTargetIdentifiers`.

That's why we're defining this and other required properties explicitly in the head, before we import anything.

image

**Comment by Arlodotexe** [comment]
Looks like we'll have to wait for WinAppSDK to ship support for `net7-windows-*` first. We can target `net6-windows`, but if left as-is our `net7.0` Uno tfm will be being picked up instead by consuming projects, if they have Uno. 

This arises from how NuGet packages pick TFMs. When the consuming Wasdk head uses `net7-windows*` but sees `net6-windows*`, instead of selecting the next lowest platform-specific TFM, it drops the platform and goes for `net7.0` instead.

This can only be fixed by waiting for the WinAppSDK to release explicit support for `net7-windows*`.

**Comment by michael-hawker** [comment]
> Looks like we'll have to wait for WinAppSDK to ship support for `net7-windows-*` first. We can target `net6-windows`, but if left as-is our `net7.0` Uno tfm will be being picked up instead by consuming projects, if they have Uno.

Looks like this change is targeting both `net6.0-windows*` and `net7.0-windows*` thought that didn't work? The main issue is the conflict from WASM between `net7.0` which is why the change to `netstandard2.0` should resolve that side. Or is the build only breaking when you have the mixed `net6.0-windows*;net7.0-windows*` for WASDK AND the `net7.0` for WASM?

I'm a bit confused based on what wasn't working for you earlier with our offline conversation...

**Comment by Arlodotexe** [comment]
> Looks like this change is targeting both net6.0-windows* and net7.0-windows* thought that didn't work? 

This was [removed](https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/125/commits/08dddc7389a2812e9af5e2776988186ba0da89a1#diff-3e274bd0c8181e6228278f26cc15080d741ef6551834c9eae565a829019975d3L4) in the second commit. 

We're still using `net6.0-windows*` for Wasdk here. If we change it to `net7.0-windows`, it breaks. Same behavior if we try to use both.

For the rest, see the original post. I've updated with our findings.

**Comment by Arlodotexe** [comment]
Thanks @nickrandolph for working with us to find the source of the issue in the first commit so we can ship this in time.

The issue in the first commit was [this line](https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/125/commits/f20aecb14dc0f62299424c2a088ff4d8a8376315#diff-454092c2ff4dfc00850478e5ab0bcf5f953e522e82c39439187190708f5a6c69R45). We're checking if the current `TargetFramework` is contained in the `WinAppSdkTargetFramework`. The check is too greedy, and picks up `net7.0` before it can pick up `net7.0-windows*`.

I've fixed the issue in efd95db19828238a88659bf839e81b9f0af0322e

File: ProjectHeads/Head.WinAppSdk.props [hunk 2]
```diff
@@ -1,10 +1,7 @@
 <Project>
-  <Import Project="$(MSBuildThisFileDirectory)\..\MultiTarget\PackageReferences\WinAppSdk.props" />
-  <Import Project="$(MSBuildThisFileDirectory)\..\MultiTarget\WinUI.Extra.props" />
-
   <PropertyGroup>
     <OutputType>WinExe</OutputType>
-    <TargetFramework>$(WinAppSdkTargetFramework)</TargetFramework>
+    <TargetFramework>$(WinAppSdkTargetFramework.Split(';')[0])</TargetFramework>
```
Comment (michael-hawker) [inline ProjectHeads/Head.WinAppSdk.props:4]: Small thing to discuss figure out later, maybe open an issue for when we add .net8 later once that ships. We probably want our Sample apps running against the latest vs. the oldest version? So, either we should grab the last one if we're always going to order them the same or should maybe have a separate property we can set for this elsewhere?

**Review by michael-hawker — APPROVED** [review]
Tested `8.0.230828-pull-194.636` packages from https://github.com/CommunityToolkit/Windows/pull/194

Worked like a charm now on Windows/WinUI3, WASM, and even Android!

Still not 100% sure if we can just add to the single class library in the Uno project vs. everything, but swapping it out from my last setup worked fine.

File: ProjectHeads/Head.WinAppSdk.props (hunk 2 above)
Comment (Arlodotexe) [reply ProjectHeads/Head.WinAppSdk.props:4]: Grabbing the last index was a bit more complicated, I figured we stick with LTS for now. Can adjust or reorder them later if needed.

**Pull request merged**